packages/x-components/src/components/__tests__/global-x-bus.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { mount } from '@vue/test-utils';
+import Vue, { ComponentOptions } from 'vue';
+import { installNewXPlugin } from '../../__tests__/utils';
+import { XComponentBusAPI } from '../../plugins/x-plugin.types';
+import GlobalXBus from '../global-x-bus.vue';
+
+function renderGlobalXBus({ listeners = {} }: RenderGlobalXBusOptions = {}): RenderGlobalXBusAPI {
+  const [, localVue] = installNewXPlugin();
+  const wrapper = mount(GlobalXBus, { listeners, localVue });
+
+  return {
+    on: wrapper.vm.$x.on.bind(wrapper.vm.$x),
+    emit: wrapper.vm.$x.emit.bind(wrapper.vm.$x)
+  };
+}
+
+describe('testing GlobalXBus component', function () {
+  it('executes a callback provided with the listeners when the event is emitted', function () {
+    const acceptedAQueryCallback = jest.fn(payload => payload);
+    const clickedColumnPickerCallback = jest.fn(payload => payload);
+    const { emit } = renderGlobalXBus({
+      listeners: {
+        UserAcceptedAQuery: acceptedAQueryCallback,
+        UserClickedColumnPicker: clickedColumnPickerCallback
+      }
+    });
+
+    emit('UserAcceptedAQuery', 'lego');
+    emit('UserClickedColumnPicker', 2);
```

Comment:
I would only emit one of those two events, and assert that the other one hasn't been called.